tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_heartbeat.c

```diff
@@ -63,7 +65,26 @@ static void sub_hb_recv_cb(const struct bt_mesh_hb_sub *sub, uint8_t hops, uint1
 	ASSERT_TRUE(sub->remaining <= SUBSCRIBE_PERIOD_SEC);
 	ASSERT_EQUAL(sub_ctx.count + 1, sub->count);
 	ASSERT_EQUAL(hops, EXPECTED_HB_HOPS);
-	ASSERT_TRUE(feat <= PUBLISH_FEATURES_RANGE);
+
+	uint16_t feature = 0;
+
+	if (bt_mesh_relay_get() == BT_MESH_RELAY_ENABLED) {
+		feature |= BT_MESH_FEAT_RELAY;
+	}
+
+	if (bt_mesh_gatt_proxy_get() == BT_MESH_GATT_PROXY_ENABLED) {
+		feature |= BT_MESH_FEAT_PROXY;
+	}
+
+	if (bt_mesh_friend_get() == BT_MESH_FRIEND_ENABLED) {
+		feature |= BT_MESH_FEAT_FRIEND;
+	}
+
+	if (bt_mesh_lpn_established()) {
+		feature |= BT_MESH_FEAT_LOW_POWER;
+	}
```

Comment:
Is this known at compile time? Would it be better to check against hardcoded value?

Reply:
This isn't known at compile time, but `CONFIG_BT_MESH_LOW_POWER` is, and as far as I can tell, that is the flag we're actually looking for here?

Reply:
I'd prefer to create a feature set dynamically. If somebody adds more tests for example related to gatt proxy then the hardcoded value will not be relevant anymore. Currently, test is tolerant to future configuration option changes.